Adds mech thruster equipment

zxaber opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a mech thrusters equipment package, which can be created at an exosuit fabricator after Advanced Mecha Tools has been researched (same node as the mech RCD). The RCS thruster package will deplete the mech's internal gas tank in order to apply thrust, at 20 moles per step. It can be toggled using the View Stats mech popout, similar to how the mech RCD is toggled.

Because they use gas from the mech's life support tank, the MK-I Ripley cannot use an RCS thruster package.

The built-in thrusters for antag-mechs (Dark Gagax, Dark Honker, Marauder and it's children) now use a variant of the RCS thruster, the ion thruster. Functionally, nothing changes except the thruster now starts enabled, and there is an ion effect when the thruster is used. Additionally, the thruster action button has been removed in favor of the View Status control setting. The thruster takes up an equipment slot, but it cannot be removed and all affected mechs have had their total equipment slots raised by one.

Added support for non-selectable equipment to still show up in the equipment controls section of the View Status popout. Thus, you cannot select the thrusters, but can still toggle them.

Added support for non-removable mech equipment.

Removed mech stompy sounds if the mech is using thrusters.
 
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Allows mechs built on a space station to have a bit of a life line in case some space leaks in or gravity dies. Takes an equipment slot, so you can't use this and also have three guns on your Durand. Uses a not-insignificant amount of gas to propel the mech and so can carry risks if you're also using that gas for your air pressure and breathing. Spawns a smoke effect when the thruster is used, which dissipates after about a second.

Nuke ops mechs should function without any noticeable differences outside the new ion particle effects when using thrusters and lack of an action button. The thrusters already don't take effect if the mech is able to walk normally (even before this PR), so there's almost never a reason to disable them; as such, I don't think they deserve an action button.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: RCS Thrusters for mechs can now be printed from an exosuit fabricator after the relevant research has been done. They'll eat up the gas in your mech's air tank, though, so don't go overboard with their use!
tweak: Mechs that use built-in thrusters (Dark Gygax, Marauder, etc) will now start with the thrusters enabled. The thruster action button is gone, but you can still toggle them in the View Status popout if you feel the need to. Note that these built-in thrusters are still ion-based, and will not deplete your air supply.
tweak: Mechs using thrusters to move will no longer make stomp sounds.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Oranges gave permission for this PR that touches antag equipment (antag mechs) during the antag freeze.

zxaber committed:
changes

zxaber committed:
Merge branch 'master' into we_gundam_now

necromanceranne commented:
I thought the ion thrusters used up power.

AnonymousNow commented:
> 
> 
> I thought the ion thrusters used up power.

Big factor. Needs a toggle if so.

zxaber commented:
They don't if you're on station or some other spot with gravity. `Process_Spacemove()` checks for that first and happily returns true in the case of gravity, bypassing any use of the thrusters. Such was already the case before this PR.

You can still toggle the thrusters, though. It's just under the equipment list in the view status popout rather than being another action button. It's in the same spot you would reload a ballistics weapon.

AnturK commented on code/game/mecha/equipment/mecha_equipment.dm:
```diff
@@ -130,6 +131,8 @@
 	return
 
 /obj/item/mecha_parts/mecha_equipment/proc/detach(atom/moveto=null)
+	if(!detachable)
```
Don't these get detached during destroy and similar, this should probably only apply on manual removal.

zxaber replied:
Far as I can tell, the destroy() proc doesn't actually call detach, and does all the work manually instead. That being said, this check probably isn't necessary since I removed the actual detach button (well, link) from equipment with detachable = FALSE anyway. I'll remove this.

zxaber committed:
changes to detach and destroy

Remove code from detach(),
Add code to Destroy()

zxaber commented:
Removed the code that halted the detach() proc for non-detachable equipment as per above, since the detach button was already modified to not show up for non-detachable equipment anyway.

Added a step in the equipment Destroy() proc that subtracts an equipment slot if a non-removable equipment is destroyed. This is so that if the thrusters of nukie mechs do ever get destroyed somehow, they can't just equip an extra weapon to the slot. Built-in means not replaceable. Since all mechs with built-in thrusters had their equipment slot number increased by one already (to hold the thrusters), this ultimately means no functional change to their number of equipment slots.

zxaber committed:
added a trailing new line

I guess this is needed now or something
